### PR TITLE
parsers: Added list joining function.

### PR DESCRIPTION
**Pull request by pepol**
- This function takes a dictionary as outputted by '_make_sections'.
 - It returns equivalent section with lines that are part of some
   top-level list joined into one.

Issue: #2
Reviewer: @JakubNvk
Cc: @Anna-Rebeka @milancerny

### CODE REVIEW

File: parser/icf.py [hunk 1]
```diff
@@ -110,3 +110,35 @@ def _make_sections(lines):
...
+
+
+def _join_lists(sections):
+  """Join lines that make up a list together.
+
+  Arguments:
+    sections - a dictionary of lists, each list representing one line.
+
+  Returns:
+    a dictionary containing lines, with each list on one line.
```
Comment (JakubNvk) [inline parser/icf.py]: @pepol can you specify this a bit more?

**Comment by JakubNvk** [comment]
please adress my comment @pepol , otherwise lgtm!

**Comment by pepol** [comment]
@JakubNvk done.

**Comment by JakubNvk** [comment]
thanks, merging..

**Pull request merged**